spec/protocol/endpoints.adoc

```diff
@@ -0,0 +1,19 @@
+= MANY Endpoints
+
+A server exposes endpoints that a message must target.
+Endpoints receive a single bytestring payload, and optionally return either an error or a bytestring payload.
+
+== Command Query Responsibility Segregation
+
+MANY separates messages into two kinds;
+
+1. Query (or read) which access data in read-only mode.
+   The app state hash should not be changed after running a query, and any queued command calls should be resolved or paused (commands SHOULD NOT be in the middle of executing).
```

Comment:
The 2nd part is not clear.

I think you mean that some data MUST NOT be accessed in both read AND write modes simultaneously, i.e., NO race condition.

Reply:
Queries can be executed with race conditions because they cannot change the memory. Commands cannot. I'll try to clarify.